lib/offsite_payments/integrations/paytm.rb

```diff
@@ -0,0 +1,475 @@
+        aes.iv = '@@@@&&&&####$$$$'
+        decrypted_k = Base64.decode64(k.to_s)
+        decrypted_k = aes.update(decrypted_k.to_s) + aes.final
+        if data.empty?
+          decrypted_data[decrypted_k] = ""
+          next
+        end
+        aes.decrypt
+        aes.key = key
+        aes.iv = '@@@@&&&&####$$$$'
+        data = Base64.decode64(data)
+        decrypted_data[decrypted_k] = aes.update(data) + aes.final
+      end
+    rescue Exception => e
+      return false
+    end
+    return decrypted_data
+  end
+  
+  
+  def new_pg_decrypt_variable(data, key)
+    aes = OpenSSL::Cipher::Cipher.new("aes-128-cbc")
+    aes.decrypt
+    aes.key = key
+    aes.iv = '@@@@&&&&####$$$$'
+    decrypted_data = nil
+    begin
+      decrypted_data = Base64.decode64(data.to_s)
+      decrypted_data = aes.update(decrypted_data) + aes.final
+    rescue Exception => e
+      return false
+    end
+    return decrypted_data
+  end
+
+
+  def new_pg_generate_salt(length)
+    salt = SecureRandom.urlsafe_base64(length*(3.0/4.0))
+    return salt.to_s
+  end
+  
+  
+  def new_pg_checksum(params, key, salt_length = 4)
+    if params.class != Hash
+      return false
+    end
+    if key.empty?
+      return false
+    end
+    
+  end
+  
+  def new_pg_verify_checksum(params, check_sum, key, salt_length = 4)
+    
+    if params.class != Hash
+      return false
+    end
+
+    if key.empty?
+      return false
+    end
+
+    if check_sum.nil? || check_sum.empty?
+      return false
+    end
+
+    generated_check_sum = nil
+    check_sum = new_pg_decrypt_variable(check_sum, key)
+
+    if check_sum == false
+      return false
+    end
+    begin
+      salt = check_sum[(check_sum.length-salt_length), (check_sum.length)]
+      keys = params.keys
+      str = nil
+      keys = keys.sort
+      keys.each do |k|
+        if str.nil?
+          str = params[k].to_s
+          next
+        end
+        str = str + '|' + params[k].to_s
+      end
+      str = str + '|' + salt
+      generated_check_sum = Digest::SHA256.hexdigest(str)
+      generated_check_sum = generated_check_sum + salt
+    rescue Exception => e
+      return false
+    end
+    
+    if check_sum == generated_check_sum
+      return true
+    else
+      return false
+    end
+  end
+	
+	
+      mattr_accessor :test_url
+      mattr_accessor :production_url
+
+      self.test_url = 'https://pguat.paytm.com/oltp-web/processTransaction'
+      self.production_url = 'https://secure.paytm.in/oltp-web/processTransaction'
+
+      def self.service_url
+         #self.production_url
+		 OffsitePayments.mode == :production ? self.production_url : self.test_url
+      end
+
+      def self.notification(post, options = {})
+        Notification.new(post, options)
+      end
+
+      def self.return(post, options = {})
+        Return.new(post, options)
+      end
+
+      def self.checksum(params, secret_key )
+			#salt = SecureRandom.urlsafe_base64(4*(3.0/4.0))
+			salt = '1234'
```

Comment:
Was this meant to stay like this or is the commented code what's intended to be used?